src/java.base/share/classes/jdk/internal/foreign/Utils.java

```diff
@@ -90,48 +110,57 @@ public static MemorySegment alignUp(MemorySegment ms, long alignment) {
     }
 
     /**
-     * This method returns a <em>raw var handle</em>, that is, a var handle that does not perform any size
-     * or alignment checks. Such checks are added (using adaptation) by {@link LayoutPath#dereferenceHandle()}.
+     * This method returns a var handle that takes a memory segment and offset (MS, long). If it has strides, it also
+     * takes a raw offset (MS, long, long), which the var handle does not perform any size or alignment checks against.
+     * Such checks are added (using adaptation) by {@link LayoutPath#dereferenceHandle()}.
      * <p>
      * We provide two level of caching of the generated var handles. First, the var handle associated
      * with a {@link ValueLayout#varHandle()} call is cached inside a stable field of the value layout implementation.
      * This optimizes common code idioms like {@code JAVA_INT.varHandle().getInt(...)}. A second layer of caching
-     * is then provided by this method: after all, var handles constructed by {@link MemoryLayout#varHandle(PathElement...)}
-     * will be obtained by adapting some raw var handle generated by this method.
+     * is then provided by this method, so different value layouts with same effects can reuse var handle instances.
```

Comment:
I believe this comment is now out of sync? (It talks about two levels of caching --- but there's only one now)

Reply:
The first level is right, it is the instance field. The static final CHM over there was not mentioned and is now removed

Reply:
I've re-read this, and I see how `cached inside a stable field` doesn't really imply use of map or something else. So yes, this text is still ok.